modular_skyrat/modules/modular_implants/code/tgui/states.dm

```diff
@@ -0,0 +1,47 @@
+/// The state used for handheld soulcatcher interaction tgui.
+/// COULD be generified, but for now, its fine. 
+/// If someone ever makes a combination alive, concious, and proximity item state like this,
+/// please do switch over.
+/// No global instance so we can pass args.
+/datum/ui_state/handheld_soulcatcher_state
+	var/obj/item/handheld_soulcatcher/linked_soulcatcher
+
+/datum/ui_state/handheld_soulcatcher_state/New(obj/item/handheld_soulcatcher/linked_soulcatcher)
+	. = ..()
+
+	src.linked_soulcatcher = linked_soulcatcher
+
+/datum/ui_state/handheld_soulcatcher_state/Destroy(force, ...)
+	linked_soulcatcher = null
+
+	return ..()
+
+#define SOULCATCHER_MAX_CATCHING_DISTANCE 7
```

Comment:
Put that at the start of the file, we do by choice defines at the start of the file as a stylistic choice, it causes less random interruptions and makes it easier to find them.

Reply:
sure